### DISCUSSION TITLE
prisma schema can use the literal type？

### DISCUSSION THREAD

**Discussion by C-Coding** [prisma/prisma#13968] [Q&A] [answered]
## Problem

for example, there are three optional value for the "Status" field
```
model User {
  id        Int      @id @default(autoincrement())
  status    Int      @default(0) @db.TinyInt //status=0; status=1; status=2
}
```

## Suggested solution
```
model User {
  id        Int      @id @default(autoincrement())
  status    Int<0,1,2>      @default(0) @db.TinyInt //status=0; status=1; status=2
}
```
that can generated file
```
/**
 * Model User
 * 
 */
export type User = {
  id: number
  status: 0|1|2
}
```

**Comment by Zeko369** [comment, marked as answer]
Why not use Enums for this?

**Reply by RogerBetter** [reply]
While enum is limited by the DBMS type as said in [https://www.prisma.io/docs/reference/database-reference/database-features#misc](url)
We who develop on premise app would love a solution that fit in any customer requested stack.
Currenty I'm building a table for every enum to solve this, but it's quite verbose.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
